sonar-ws/src/main/java/org/sonarqube/ws/client/PostRequest.java

```diff
@@ -17,55 +17,53 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-
 package org.sonarqube.ws.client;
 
-import org.junit.rules.ExternalResource;
-
+import java.io.File;
+import java.util.LinkedHashMap;
 import java.util.Map;
 
-public final class MockHttpServerInterceptor extends ExternalResource {
+/**
+ * @since 5.3
+ */
+public class PostRequest extends BaseRequest<PostRequest> {
 
-  private MockHttpServer server;
+  private final Map<String, Part> parts = new LinkedHashMap<>();
 
-  @Override
-  protected final void before() throws Throwable {
-    server = new MockHttpServer();
-    server.start();
+  public PostRequest(String path) {
+    super(path);
   }
 
   @Override
-  protected void after() {
-    server.stop();
+  public Method getMethod() {
+    return Method.POST;
   }
 
-  public MockHttpServerInterceptor stubResponseBody(String body) {
-    server.doReturnBody(body);
+  public PostRequest setPart(String name, Part part) {
```

Comment:
why ? It replaces existing key and method name for string params is already `setParam()`.